osu.Game/Overlays/Direct/FilterControl.cs

```diff
@@ -0,0 +1,273 @@
+                set
+                {
+                    ruleset = value;
+                    icon.Icon = Ruleset.CreateInstance().Icon;
+                }
+            }
+
+            private readonly Bindable<RulesetInfo> bindable;
+
+            private void Bindable_ValueChanged(RulesetInfo obj)
+            {
+                icon.FadeTo(Ruleset.ID == obj?.ID ? 1f : 0.5f, 100);
+            }
+
+            public RulesetToggleButton(Bindable<RulesetInfo> bindable, RulesetInfo ruleset)
+            {
+                this.bindable = bindable;
+                AutoSizeAxes = Axes.Both;
+
+                Children = new[]
+                {
+                    icon = new TextAwesome
+                    {
+                        Origin = Anchor.TopLeft,
+                        Anchor = Anchor.TopLeft,
+                        TextSize = 32,
+                    }
+                };
+
+                Ruleset = ruleset;
+                bindable.ValueChanged += Bindable_ValueChanged;
+                Bindable_ValueChanged(bindable.Value);
+                Action = () => bindable.Value = Ruleset;
+            }
+
+            protected override void Dispose(bool isDisposing)
+            {
+                if (bindable != null)
+                    bindable.ValueChanged -= Bindable_ValueChanged;
+                base.Dispose(isDisposing);
+            }
+        }
+
+        private class DisplayStyleToggleButton : ClickableContainer
+        {
+            private readonly TextAwesome icon;
+            private readonly PanelDisplayStyle style;
+            private readonly Bindable<PanelDisplayStyle> bindable;
+
+            public DisplayStyleToggleButton(FontAwesome icon, PanelDisplayStyle style, Bindable<PanelDisplayStyle> bindable)
+            {
+                this.bindable = bindable;
+                this.style = style;
+                Size = new Vector2(SlimEnumDropdown<DirectTab>.HEIGHT);
+
+                Children = new Drawable[]
+                {
+                    this.icon = new TextAwesome
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Icon = icon,
+                        TextSize = 18,
+                        UseFullGlyphHeight = false,
+                        Alpha = 0.5f,
+                    },
+                };
+
+                bindable.ValueChanged += Bindable_ValueChanged;
+                Bindable_ValueChanged(bindable.Value);
+                Action = () => bindable.Value = this.style;
+            }
+
+            private void Bindable_ValueChanged(PanelDisplayStyle style)
+            {
+                icon.FadeTo(style == this.style ? 1.0f : 0.5f, 100);
+            }
+
+            protected override void Dispose(bool isDisposing)
+            {
+                bindable.ValueChanged -= Bindable_ValueChanged;
+            }
+        }
+    }
+
+    public class ResultCounts

```

Comment:
either move these to their own file or nest inside class (we're trying to maintain one object per file)